test/sequential/test-fs-watch.js

```diff
@@ -126,3 +128,21 @@ assert.throws(function() {
   w.stop();
 }, TypeError);
 oldhandle.stop(); // clean up
+
+assert.throws(function() {
+  fs.watch('non-existent-file');
+}, function(err) {
+  assert(err);
+  assert(/non-existent-file/.test(err));
+  assert.equal(err.filename, 'non-existent-file');
+  return true;
+});
+
+var watcher = fs.watch(__filename);
+watcher.on('error', function(err) {
```

Comment:
Instead of creating `watcherErrorCaught`, you can just wrap this function in `common.mustCall()`. That way you can ensure it was called, and fail the test if it's not.